File: Day01/one.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error opening input file")
		os.Exit(1)
	}
	defer file.Close()
	var inputarr []int
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		x, _ := strconv.Atoi(scanner.Text())
		inputarr = append(inputarr, x)
	}
	lastindex := len(inputarr) - 2
	incCount := 0
	for n := 0; n <= lastindex; n++ {
		if inputarr[n] < inputarr[n+1] {
			incCount++
		}
	}
	fmt.Println(incCount)
}

File: Day01/one.py
f = open('input.txt')
inputlist = f.read().splitlines()
f.close()
inccount = 0
slidingwindowinc = 0
slidingwindow = []
for n in range(0, len(inputlist) - 1):
    if  n < len(inputlist) - 2:
        slidingwindow.append(int(inputlist[n]) + int(inputlist[n+1]) + int(inputlist[n+2]))
    if int(inputlist[n]) < int(inputlist[n+1]):
        inccount += 1
for x in range(0, len(slidingwindow) - 1):
    if slidingwindow[x] < slidingwindow[x+1]:
        slidingwindowinc += 1

print(inccount)
print(slidingwindowinc)
